data/hai/hai reveal 1 intro.txt

```diff
@@ -0,0 +1,1002 @@
+			choice
+				`	"Yes."`
+					goto continue
+			label innocent
+			`	"Hah!" she booms, one deep short bark of a laugh. "You and everyone else, <last>, and don't go trying to sell me the innocent question line. You didn't single me out cause I was pretty."`
+			`	She squints at you and leans back in her seat slightly. "That said, if you're asking me and not any of your new 'friends' in the government... maybe you're alright in my books after all."`
+				goto continue
+			label practical
+			`	Her eyes narrow, but she dips her head in a very small motion of assent. "Well, you're not stupid, I'll give you that. You're trying to find out what most other people are, aren't you? You want to know about the Hai."`
+			choice
+				`	"Yes."`
+			label continue
+			`	She heaves a sigh and lowers her voice until you're forced to lean in. "You can call me Captain Casey, and I will tell you what you want, but then you can leave, alright? I like my solitude.`
+			`	"First, I shouldn't be telling you this. Generally speaking they don't like to draw attention to themselves. Second though, forget the fearmongering. I make trips up to their space every so often, and there aren't a nicer bunch of people anywhere, even if they are a bit furry. The Hai are peaceful and welcoming of outsiders. They've got themselves a bit of an anarchist faction, that's the footage you've seen, but they ain't no danger to humanity. You should see them for yourself."`
+			`	She pulls out her data pad and flicks some coordinates over to you. "Follow those and you'll get there. Keep an eye out for the wormhole."`
+			choice
+				`	"Thank you."`
+					goto wrap
+				`	"The wormhole?"`
+			`	She shakes her head. "See for yourself."`
+			label wrap
+			`	Taking a sip from her mug she nods slightly, once, in your direction. "Intentionally or not, you did me and a lot of other merchants a solid in that war, Captain. I respect that, but I'll thank you to leave me in peace if you see me again."`
+			`	Only now, as you move to depart, do you spot the wrist-triggered needlegun strapped to the arm of her drinking hand that would've been trained on you the entire conversation.`
+				accept
+			
+
+#This branch is if you have met the Hai but not the Unfettered
+	#Goto Stop 3 Wah Yoot (Darkcloak) and pick up at Tour 4 mission.
+mission "Hai Witness Unfettered [1]"
+	name "Witness the Unfettered"
+	description "It turns out this conflict has some truth to it. Go see where it's happening."
+	landing
+	source "Allhome"
+	destination "Darkcloak"
+	to offer
+		or
+			has "ZZ Event Hai Secret Leaks: RSH: done"
+			has "ZZ Event Hai Secret Leaks: FWH: done"
+
+
+
+# This fires if you haven't met the Hai, to trigger the entire tour.
+# It must happen after the first contact mission.
+	#Goto Stop 1 Imo Dep (Stonebreak)
+	#This follows immediately after the First Contact text - no other convo is required.
```

Comment:
Mmmm, you are explicitly sent to Allhome to trigger this, but I see your point.

To what extent does this need to be fully idiot-proofed before merging though?
Both this and your point about restructuring the named conversations bit (which I thought was a good things as a demonstration of the functionality) feel like they'd be better stress-tested in their own PR after it's merged when we start seeing what people do to break it.

Reply:
That whole "Merge things and fix them latter as people break them," is exactly something I've said I want to avoid. Let's get this right the first time. Breaking things is exactly what the review phase is for.

> To what extent does this need to be fully idiot-proofed before merging though?

I wouldn't call it idiot proofing as much as I would call it doing things as the game has done them already. There isn't anywhere that I can recall where the game has so strictly required you to go to exactly where you need to in order to have everything work properly, and if you land anywhere else you break thing. As I mentioned in another review comment somewhere (I think), the player has a jump drive at this point. It may be that they don't even enter Hai space through the wormhole and therefore Allhome wouldn't be the first planet that come across the land on.

Reply:
> I think an alternate first contact at this point is a perfectly reasonable expectation for this PR.

I think it would be perfectly reasonable if someone had got to this sooner than _two years_ after it opened. As it stands, no. It can happen after when there are fewer moving parts involved.